src/Frontend/Core.hs

```diff
@@ -144,7 +144,7 @@ makeFrame :: (ActionPersist r m, ActionUserHandler m, MonadError ActionExcept m,
 makeFrame p = do
   isli <- isLoggedIn
   if | not isli && isPrivatePage p -> redirect "/login"
-     | isPrivatePage p             -> flip Frame p <$> currentUser
+     | isli     || isPrivatePage p -> flip Frame p <$> currentUser
```

Comment:
This is exactly the place where `currentMaybeUser` and a case would be better. (no need to did yet though)